packages/compass-aggregations/src/components/stage-editor-toolbar/stage-operator-select.jsx

```diff
@@ -1,61 +1,88 @@
 import PropTypes from 'prop-types';
-import Select from 'react-select-plus';
 import { connect } from 'react-redux';
 import { changeStageOperator } from '../../modules/pipeline-builder/stage-editor';
-import SelectOptionWithTooltip from './select-option-with-tooltip/select-option-with-tooltip';
 import { filterStageOperators } from '../../utils/stage';
 
-import styles from './stage-operator-select.module.less';
+import { Combobox, ComboboxOption, css, cx, spacing } from '@mongodb-js/compass-components';
+import { isAtlasOnly } from '../../utils/stage';
 
-/**
- * Select from a list of stage operators.
- */
+const inputWidth = spacing[7] * 2;
+const descriptionWidth = spacing[5] * 14;
+const stageNameWidth = spacing[4] * 8;
+const dropdownWidth = stageNameWidth + descriptionWidth;
 
-export class StageOperatorSelect extends PureComponent {
-  static propTypes = {
-    index: PropTypes.number.isRequired,
-    stages: PropTypes.array.isRequired,
-    selectedStage: PropTypes.string,
-    onChange: PropTypes.func.isRequired,
-    isDisabled: PropTypes.bool,
+const comboboxStyles = css({
+  marginLeft: spacing[2],
+  width: inputWidth,
+  '& [role="combobox"]': {
+    height: spacing[4] - 2 // match other xs controls
   }
+});
 
-  /**
-   * Called when the stage operator is selected.
-   * @param {String} name The name of the stage operator.
-   * @returns {void}
-   */
-  onStageOperatorSelected = (name) => {
-    this.props.onChange(this.props.index, name);
-  }
-
-  /**
-   * Render the stage operator select component.
-   *
-   * @returns {Component} The component.
-   */
-  render() {
-    return (
-      <div className={styles['stage-operator-select']}>
-        <Select
-          optionComponent={SelectOptionWithTooltip}
-          simpleValue
-          searchable
-          openOnClick
-          openOnFocus
-          clearable={false}
-          disabled={this.props.isDisabled}
-          className={styles['stage-operator-select-control']}
-          options={this.props.stages}
-          value={this.props.selectedStage}
-          onChange={this.onStageOperatorSelected}
-        />
-      </div>
-    );
-  }
+function comboboxOptionStyles(stage) {
+  return css({
+    '&::after': {
+      content: JSON.stringify(
+        (isAtlasOnly(stage.env) ? 'Atlas only. ' : '') +
+        stage.description),
+      width: descriptionWidth
+    },
+  });
 }
 
+const portalStyles = css({
+  '> div': {
+    width: dropdownWidth,
+    marginLeft: (dropdownWidth / 2) - (inputWidth / 2) // realigns the dropdown with the input
+  },
+});
+
+export const StageOperatorSelect = ({
+  onChange,
+  index,
+  selectedStage,
+  stages
+}) => {
+  const onStageOperatorSelected = useCallback((name) => {
+    onChange(index, name);
+  }, [onChange, index]);
+
+  const optionStyleByStageName = useMemo(() => {
+    return Object.fromEntries(stages.map((stage) => [stage.name, comboboxOptionStyles(stage)]))
+  }, [stages])
+
+  return <Combobox value={selectedStage}
```

Comment:
Fine as is, however if we want to make this scroll a bit nicer with the aggregations content we could pass a scrollContainer reference to the aggregation pipeline window.
https://www.mongodb.design/component/combobox/documentation/

I was thinking a `ScrollContainerProvider` with a scroll container in the context might be a useful thing for places like this, and with the menus for editable documents. I'll probably add it for https://jira.mongodb.org/browse/COMPASS-6271 so maybe it can get some use there.